### PR TITLE
[BE] Follow and Unfollow

### PR DESCRIPTION
**Pull request by Vali-R**
## Asana Link
https://app.asana.com/0/1202860323783357/1202860323783412/f

 ## Commands
```
cd backend
npm start
```
**Please send PUT request to `http://localhost:YOUR_BACKEND_PORT/follows`**

## Pre-conditions

-**It expects `id` param, it is the id of the user you are gonna follow or unfollow**

## Expected Output

- [x]  **You should be able to follow**
- [x] **If you are already following the user, it should unfollow** 
- [x] **It should add the activity if you follow or unfollow** 

## Documentation
- Postman APIs - https://grey-firefly-150180.postman.co/workspace/e-learning~b9d71430-452d-47cc-8c64-e2b2e960bd99/example/22825265-04fdb3a0-77bc-42fe-8dae-153965b37c87

## Notes
**Backend integration only**

 ## Screenshots

image

image

### CODE REVIEW

File: backend/migrations/20220906063928-follows-flag.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"use strict";
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
```
Comment (escbooster12-dev) [inline backend/migrations/20220906063928-follows-flag.js:4]: - when you generate migration. please consider proper filenaming.

File: backend/routes/feed.js [hunk 2]
```diff
@@ -19,4 +20,6 @@ router.get("/user/info/:id", getUserInfo);
 
 router.get("/users", getAllUsersInfo);
 
+router.put("/user/follow_and_unfollow", postfollowAndUnfollow);
```
Comment (escbooster12-dev) [inline backend/routes/feed.js:23]: - change this to parameter type. e.g. `/comments/:id`

**Comment by escbooster12-dev** [comment]
- it keeps inserting although `follower_id` and `following_id` is already existing.
- expected result should be, if combination of `follower_id` and `following_id` are not exist, insert a new row. Else if there is already exsiting `follower_id:1` and `following_id:2` update the column `flag` and don't insert a new one (refer to image below)
image

**Comment by Vali-R** [comment]
> * it keeps inserting although `follower_id` and `following_id` is already existing.
> * expected result should be, if combination of `follower_id` and `following_id` are not exist, insert a new row. Else if there is already exsiting `follower_id:1` and `following_id:2` update the column `flag` and don't insert a new one (refer to image below)
>   image

I do see this issue on my end, could you try to pull the latest change again please
image

**Comment by escbooster12-dev** [comment]
@Vali-R 
>I do see this issue on my end, could you try to pull the latest change again please

I am the latest commit. I can see in your screenshot the same issue. There should be no duplicate `1` and `3`
image
image

**Comment by escbooster12-dev** [comment]
- it does not insert new row if table is empty

https://user-images.githubusercontent.com/73863685/188796519-bc246d72-1ab9-458a-9c0c-a5dbd87ba249.mp4

**Review by escbooster12-dev — APPROVED** [review]

**Pull request merged**